src/main/java/com/google/sps/servlets/NewRequestServlet.java

```diff
@@ -92,37 +99,50 @@ public void setFirebaseAuth(FirebaseAuthentication firebaseAuth) {
       userId = firebaseAuth.getUserIdFromIdToken(request.getParameter("idToken"));
     } catch (FirebaseAuthException e) {
       response.sendError(HttpServletResponse.SC_UNAUTHORIZED, e.getMessage());
       return;
     }
 
     DeliverySlotManager slotManager = new DeliverySlotManager();
+    DeliverySlot deliverySlot = null;
     try {
-      slotManager.createSlot(request.getParameter(DeliverySlotParameters.DELIVERY_DATE.label),
-          request.getParameter(DeliverySlotParameters.TIMEZONE_OFFSET_MINUTES.label),
-          request.getParameter(DeliverySlotParameters.START_TIME.label),
-          request.getParameter(DeliverySlotParameters.END_TIME.label),
-          userId);
-
-      if (request.getParameter(DeliverySlotParameters.LAT.label) == null || 
-        request.getParameter(DeliverySlotParameters.LNG.label) == null) {
+      // Get the delivery slot details as parameters from the form in deliveryRequest.html.
+      long timezoneOffsetMiliseconds = parseInt(request.getParameter(DeliverySlotParameters.TIMEZONE_OFFSET_MINUTES.label)) * 60000;
+      DateTimeFormatter timeFormatter = DateTimeFormatter.ofPattern(timeFormat);
+      long startTimeInMiliseconds = LocalTime.parse(request.getParameter(DeliverySlotParameters.START_TIME.label), timeFormatter)
+          .toNanoOfDay() / 1000000 + timezoneOffsetMiliseconds;
+      long endTimeInMiliseconds = LocalTime.parse(request.getParameter(DeliverySlotParameters.END_TIME.label), timeFormatter)
+          .toNanoOfDay() / 1000000 + timezoneOffsetMiliseconds;
+      SimpleDateFormat dateFormatter = new SimpleDateFormat(dateFormat);
+      Date deliveryDate = dateFormatter.parse(request.getParameter(DeliverySlotParameters.DELIVERY_DATE.label));
+
+      if (request.getParameter(DeliverySlotParameters.LAT.label) == null ||
+          request.getParameter(DeliverySlotParameters.LNG.label) == null) {
         // the user has set the address himself
-        slotManager.setSlotStartingPoint(request.getParameter(DeliverySlotParameters.START_ADDRESS.label));
+        deliverySlot = slotManager.createDeliverySlot(deliveryDate,
+            startTimeInMiliseconds,
+            endTimeInMiliseconds,
+            userId,
+            request.getParameter(DeliverySlotParameters.START_ADDRESS.label));
       } else {
         // the starting point is the current location of the user
-        slotManager.setSlotStartingPoint(request.getParameter(DeliverySlotParameters.LAT.label), 
-          request.getParameter(DeliverySlotParameters.LNG.label));
+        deliverySlot = slotManager.createDeliverySlot(deliveryDate,
+            startTimeInMiliseconds,
+            endTimeInMiliseconds,
+            userId,
+            parseDouble(request.getParameter(DeliverySlotParameters.LAT.label)),
+            parseDouble(request.getParameter(DeliverySlotParameters.LNG.label)));
       }
-    } catch (BadRequestException e) {
+    } catch (BadRequestException | NumberFormatException | ParseException e) {
```

Comment:
nit: Big try/catch blocks are generally seen as poor form. Does it have to be that wide of a net or can they be multiple smaller try/catch blocks catching fewer exceptions?

Reply:
I can split it so that the parse exceptions are catched separately, but I used the big try/catch block to avoid declaring the local variables outside of the block (e.g.` long timezoneOffsetMiliseconds`).